internal/pkg/db/db.go

```diff
@@ -90,15 +101,131 @@ func (backend *Backend) AddDocument(document *sbom.Document) error {
+	switch documents, getDocsErr := backend.GetDocumentsByID(id); {
+	case getDocsErr != nil:
+		err = fmt.Errorf("querying documents: %w", getDocsErr)
+	case len(documents) == 0:
+		doc = nil
+	case len(documents) > 1:
+		err = fmt.Errorf("%w %s", errMultipleDocuments, id)
+	default:
+		doc = documents[0]
+	}
+
+	return doc, err
+}
+
+func (backend *Backend) GetDocumentByIDOrAlias(id string) (*sbom.Document, error) { //nolint:varnamelen
+	document, err := backend.GetDocumentByID(id)
+	if err != nil {
+		return nil, fmt.Errorf("document could not be retrieved: %w", err)
+	}
+
+	if document == nil {
+		switch documents, getDocErr := backend.GetDocumentsByAnnotation(AliasAnnotation, id); {
+		case getDocErr != nil:
+			err = fmt.Errorf("document could not be retrieved: %w", getDocErr)
+		case len(documents) == 0:
+			document = nil
+		case len(documents) > 1:
+			err = fmt.Errorf("%w %s", errMultipleDocuments, id)
+		default:
+			document = documents[0]
+		}
+	}
+
+	return document, err
+}
+
+func (backend *Backend) GetDocumentsByIDOrAlias(ids ...string) ([]*sbom.Document, error) {
+	if len(ids) == 0 {
+		documents, err := backend.GetDocumentsByID()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get documents: %w", err)
+		}
+
+		return documents, nil
+	}
+
+	documents := make([]*sbom.Document, len(ids))
+
+	for idx := range ids {
+		document, err := backend.GetDocumentByIDOrAlias(ids[idx])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get documents by ID: %w", err)
+		}
+
+		if document == nil {
+			return []*sbom.Document{}, nil
+		}
+
+		documents[idx] = document
+	}
+
+	return documents, nil
+}
+
+func (backend *Backend) FilterDocumentsByTag(documents []*sbom.Document, tags ...string) ([]*sbom.Document, error) {
+	taggedDocuments, err := backend.GetDocumentsByAnnotation(TagAnnotation, tags...)
 	if err != nil {
-		backend.Logger.Warn("Document could not be retrieved", "id", id, "err", err)
+		return nil, fmt.Errorf("failed to get documents by tag: %w", err)
+	}
+
+	taggedDocumentIDs := make([]string, len(taggedDocuments))
+	for idx := range taggedDocuments {
+		taggedDocumentIDs[idx] = taggedDocuments[idx].GetMetadata().GetId()
+	}
+
+	filteredDocuments := []*sbom.Document{}
+
+	for _, doc := range documents {
+		if slices.Contains(taggedDocumentIDs, doc.GetMetadata().GetId()) {
+			filteredDocuments = append(filteredDocuments, doc)
+		}
+	}
+
+	documents = filteredDocuments
+
+	return documents, nil
+}
+
+func (backend *Backend) SetAlias(documentID, alias string) (err error) {
+	if err := backend.validateNewAlias(alias); err != nil {
+		return fmt.Errorf("failed to set alias: %w", err)
+	}
+
+	docAlias, err := backend.GetDocumentUniqueAnnotation(documentID, AliasAnnotation)
+	if err != nil {
+		return fmt.Errorf("failed to set alias: %w", err)
+	}
+
+	if err := backend.RemoveAnnotations(documentID, AliasAnnotation, docAlias); err != nil {
```

Comment:
Do we want to automatically remove any existing aliases? 

If so, we need to at least log what we did since there is no mention that a document can only have one alias.

Reply:
Another option would be a confirmation dialog, but when a user assigns an alias to a document, the command is explicit enough that I feel the intent is clear enough to not warrant that confirmation. That's why I went with just replacing it as a default.

I'll log the action as a debug log and try to make it clear in the documentation that an SBOM document can have only a single alias and that alias must be globally unique.

Reply:
My preference would be not to prompt the user if at all avoidable. I prefer the pattern of having a `--force` flag and warning the user if it already exists as well as recommending the `--force` flag if that's what they intended

Reply:
I added the `--force` flag to `alias set` as suggested. And the error outputted when attempting to set an alias on a document that already has one mentions the `--force` flag as suggested.